aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -0,0 +1,349 @@
+     * configuration methods.
+     */
+    public static final class Builder {
+        private DataStoreErrorHandler dataStoreErrorHandler;
+        private DataStoreConflictHandler dataStoreConflictHandler;
+        private Long syncIntervalInMinutes;
+        private Integer syncMaxRecords;
+        private Integer syncPageSize;
+
+        private Builder() {
+            this.dataStoreErrorHandler = DefaultDataStoreErrorHandler.instance();
+            this.dataStoreConflictHandler = ApplyRemoteConflictHandler.instance(dataStoreErrorHandler);
+        }
+
+        /**
+         * A handler that will be invoked whenever there is a conflict between two model instances,
+         * one in the local store, and one from the remote server, as received from a sync operation.
+         * @param dataStoreConflictHandler A handler to invoke upon sync conflicts
+         * @return Current builder
+         */
+        @NonNull
+        public Builder dataStoreConflictHandler(@NonNull DataStoreConflictHandler dataStoreConflictHandler) {
+            this.dataStoreConflictHandler = Objects.requireNonNull(dataStoreConflictHandler);
+            return Builder.this;
+        }
+
+        /**
+         * Sets a handler function to be applied when the DataStore encounters an unrecoverable error
+         * in one of its ongoing background operations (model synchronization).
+         * @param dataStoreErrorHandler A handler for unrecoverable background errors
+         * @return Current builder instance
+         */
+        @NonNull
+        public Builder dataStoreErrorHandler(@NonNull DataStoreErrorHandler dataStoreErrorHandler) {
+            this.dataStoreErrorHandler = Objects.requireNonNull(dataStoreErrorHandler);
+            return Builder.this;
+        }
+
+        /**
+         * Sets the duration of time after which delta syncs will not be preferred over base syncs.
+         * @param syncIntervalInMinutes The amount of time that must elapse for delta syncs to not be considered
+         * @return Current builder instance
+         */
+        @NonNull
+        public Builder syncIntervalInMinutes(@IntRange(from = 0) Long syncIntervalInMinutes) {
+            //Only set this value if the incoming value is null
+            if (this.syncIntervalInMinutes == null) {
+                this.syncIntervalInMinutes = syncIntervalInMinutes;
+            }
+            return Builder.this;
+        }
+
+        /**
+         * Sets the maximum number of records, from the server, to process from a sync operation.
+         * @param syncMaxRecords Max number of records client will consumer from server at one time
+         * @return Current builder instance
+         */
+        @NonNull
+        public Builder syncMaxRecords(@IntRange(from = 0) Integer syncMaxRecords) {
+            //Only set this value if the incoming value is null
+            if (this.syncMaxRecords == null) {
```

Comment:
This is still a little spooky. If do:
```
builder.syncMaxRecords(1000)
    .syncMaxRecords(500)
    .build();
```
The value is 1000 even though I'd probably expect it to be 500...

(Same for others with this pattern.)

Reply:
Yeah...i agree, it feels wonky. I’ll review this tomorrow just to make sure. The struggle here is trying to come up with a mechanism that we can use to determine which values have precedence over the other. I would imagine we want to create the object with defaults, then read from file, then use anything passed in by the user. 

I’ll take another look at it tomorrow.